### PR TITLE
Makes miniguns overheat on action

### PR DESCRIPTION
**Pull request by Razharas**
Makes miniguns overheat on action
Fix https://github.com/tgstation/-tg-station/pull/17629
Not a feature

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/rechargable_magazine.dm [hunk 1]
```diff
@@ -167,6 +178,14 @@
 	else
 		qdel(src)
 
+/obj/item/weapon/gun/projectile/minigun/shoot_live_shot(mob/living/user as mob|obj, pointblank = 0, mob/pbtarget = null, message = 1)
+	if(ammo_pack)
+		if(ammo_pack.overheat < ammo_pack.overheat_max)
+			. = ..()
+			ammo_pack.overheat++
+		else
+			user << "The gun's heat sensor locked the trigger to prevent lense damage."
```
Comment (coiax) [inline code/modules/projectiles/guns/projectile/rechargable_magazine.dm]: lens

**Comment by Supermichael777** [comment]
This will never come up in competent use of this device. drop it to 40

**Comment by coiax** [comment]
Also, generally people balance "heating" by having the weapon overheating and cooling down slower than the natural rate when we reach a certain threshold.

File: code/modules/projectiles/guns/projectile/rechargable_magazine.dm [hunk 2]
```diff
@@ -65,11 +65,22 @@
 	w_class = 5
 	var/obj/item/weapon/gun/projectile/minigun/gun = null
 	var/armed = 0 //whether the gun is attached, 0 is attached, 1 is the gun is wielded.
+	var/overheat = 0
+	var/overheat_max = 40
+	var/heat_diffusion = 1
 
 /obj/item/weapon/minigunpack/New()
 	gun = new(src)
+	SSobj.processing += src
```
Comment (Iamgoofball) [inline code/modules/projectiles/guns/projectile/rechargable_magazine.dm]: fast process this bad boy and crank up the values, and put a bar sprite on the minigun showing the overheat status

File: code/modules/projectiles/guns/projectile/rechargable_magazine.dm [hunk 3]
```diff
@@ -167,6 +178,14 @@
 	else
 		qdel(src)
 
+/obj/item/weapon/gun/projectile/minigun/shoot_live_shot(mob/living/user as mob|obj, pointblank = 0, mob/pbtarget = null, message = 1)
+	if(ammo_pack)
+		if(ammo_pack.overheat < ammo_pack.overheat_max)
+			. = ..()
+			ammo_pack.overheat++
+		else
+			user << "The gun's heat sensor locked the trigger to prevent lens damage."
```
Comment (Iamgoofball) [inline code/modules/projectiles/guns/projectile/rechargable_magazine.dm]: nah, let us start doing burn damage if its overheated too far

Comment (Razharas) [inline code/modules/projectiles/guns/projectile/rechargable_magazine.dm]: It should be done in process then

File: code/modules/projectiles/guns/projectile/rechargable_magazine.dm (hunk 2 above)
Comment (Razharas) [inline code/modules/projectiles/guns/projectile/rechargable_magazine.dm]: No idea what is fast process and how do status bars are done

**Comment by optimumtact** [comment]
Lets start with this and tweak as we go

**Comment by KorPhaeron** [comment]
40 shots before overheat at 10 damage each=400 damage

1 lasergun with 12 shots at 20 each is 240. Carrying two laserguns is probably better than using a gatling now, since they have no slowdown and dont take up your entire inventory. I think it charges slower than a lasergun as well.

**Pull request merged**